server/src/main/java/org/apache/druid/segment/realtime/appenderator/TaskLockCallback.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment.realtime.appenderator;
+
+import org.joda.time.Interval;
+
+/**
+ * This interface provides a callback mechanism to interact with TaskLockbox for releasing interval locks when
+ * the segments are handed off.
+ */
+public interface TaskLockCallback
+{
+  /**
+   * Releases the lock for the given interval.
+   *
+   * @param interval interval for which the lock needs to be released
+   */
+  default void releaseLock(Interval interval)
```

Comment:
This is a new functional interface, we shouldn't need to provide a default implementation for the method.